contracts/staking/ServiceStakingBase.sol

```diff
@@ -541,38 +593,78 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
         emit ServiceUnstaked(serviceId, msg.sender, multisig, nonces, reward, tsStart);
     }
 
-    /// @dev Calculates service staking reward at current timestamp.
-    /// @param serviceId Service Id.
-    /// @return reward Service reward.
-    function calculateServiceStakingReward(uint256 serviceId) external view returns (uint256 reward) {
-        // Get current service reward
-        ServiceInfo memory sInfo = mapServiceInfo[serviceId];
-        reward = sInfo.reward;
+    /// @dev Evicts the service due to its extended inactivity.
+    function evict(uint256 serviceId) external {
```

Comment:
As part of checkpoint we should evict all that match the eviction condition imo.

Reply:
This is a bigger change that can be applied in a next PR IMO.